src/main/java/io/aiven/kafka/connect/common/output/avro/AvroConfig.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.avro;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.file.CodecFactory;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+final class AvroConfig extends AbstractConfig {
+
+    private static final String GROUP_AVRO = "Avro";
+    private static final String AVRO_CODEC = "avro.codec";
+    private static final String DEFAULT_AVRO_CODEC = "null";
+
+    AvroConfig(final ConfigDef configDef, final Map<?, ?> originals) {
+        super(configDef, originals);
+    }
+
+    CodecFactory codecFactory() {
+        final String codecName = originals()
+            .getOrDefault(AVRO_CODEC, DEFAULT_AVRO_CODEC)
+            .toString();
+        return CodecFactory.fromString(codecName);
+    }
+
+    private static ConfigDef createAvroConfigDefinition() {
+        final ConfigDef configDef = new ConfigDef();
+        configDef.define(
+            AVRO_CODEC,
+            ConfigDef.Type.STRING,
+            DEFAULT_AVRO_CODEC,
+            new CodecValidator(),
+            ConfigDef.Importance.MEDIUM,
+            "Avro Container File codec.",
+            GROUP_AVRO,
+            0,
+            ConfigDef.Width.NONE,
+            AVRO_CODEC
+        );
```

Comment:
Isn't this config conflicting with common `compression` config? What will happen when both are defined? Maybe we need a global validation to handle this.

Reply:
I can add a test. The Avro codec is for the blocks in the Container File. The compression is for the actual file produced and I think nothing should prevent compressing the Container Files.

Reply:
Unit test added, the integration test can be viewed from `gcs` connector repo: https://github.com/aiven/gcs-connector-for-apache-kafka/pull/267/files#diff-01f702c97b3e718c881c2de527e99b7517cb4929838e0e38c26523963956a813R211